django_access_logger/tests/test.py

```diff
@@ -0,0 +1,53 @@
+from django.test import TestCase
+import logging
+# Let's try to do this: https://github.com/selwin/django-user_agents/blob/master/django_user_agents/tests/urls.py
+
+access_log = logging.getLogger("django.advanced_access_logs")
+
+
+class AccessLogsMiddlewareTest(TestCase):
+    def test_headers(self):
+        resp = self.client.get('/basic_response/')
+        request = resp.wsgi_request
+        self.assertIn('aalm_timestamp_started', request.META)
+        self.assertIn('aalm_original_request', request.META)
+        self.assertIn('aalm_exceptions', request.META)
+
+    def test_log(self):
+        with self.assertLogs(access_log) as cm:
+            self.assertEquals(len(cm.records), 0)
+            self.client.get('/basic_response/')
+            self.assertEquals(len(cm.records), 1)
+            log = cm.records[0]
+        self.assertEquals(log.message, 'request processed')
+        expected_props = (
+            'duration',
+            'x_client_address',
+            'errors',
+            # that part is kindda wild
+            'request.content.mime_type',
+            'request.content.size',
+            'request.content.value',
+            'request.headers.cookie',
+            'request.http_version',
+            'request.method',
+            'request.path',
+            'response.content.mime_type',
+            'response.content.size',
+            'response.content.value',
+            'response.headers.content-type',
+            'response.status'
+        )
+        self.assertTrue(all(hasattr(log, p) for p in expected_props))
```

Comment:
On regression this assertion will produce `AssertionError: False is not true`, which is not of much help to locate the missing property.
Could you rather loop over `expected_props` and try to read each single property with `getattr(log, p)`? It would output something like `AttributeError: 'object' object has no attribute 'duration'` on error.

Reply:
There you go